### ISSUE TITLE
How can it be guaranteed that Adam updates on condition that BatchNorm updates?

### ISSUE THREAD

**Issue by haichaoyu** [tensorlayer/SRGAN#53]
Where: Line 107 - 110, main.py

To force BatchNorm layer to update, usually it is
        
    update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
        with tf.control_dependencies(update_ops):
    # Ensures that we execute the update_ops before performing the train_op
    train_op = tf.train.GradientDescentOptimizer(0.01).minimize(loss)

How can it be guaranteed in your code?

Thanks!
